packages/office-ui-fabric-react/src/utilities/keytips/KeytipManager.ts

```diff
@@ -0,0 +1,430 @@
+import { KeytipLayerBase } from '../../KeytipLayer';
+import { KeytipTree } from './KeytipTree';
+import { IKeytipTreeNode } from './IKeytipTreeNode';
+import { IKeytipProps } from '../../Keytip';
+import {
+  IKeySequence,
+  convertSequencesToKeytipID,
+  transitionKeysContain,
+  IKeytipTransitionKey,
+  fullKeySequencesAreEqual,
+  getDocument,
+  replaceElement,
+  findIndex,
+  mergeOverflowKeySequences,
+  Async
+} from '../../Utilities';
+import { constructKeytipExecuteTargetFromId } from './KeytipUtils';
+
+export interface IUniqueKeytip {
+  uniqueID: string;
+  keytip: IKeytipProps;
+}
+
+export class KeytipManager {
+  private static _instance: KeytipManager = new KeytipManager();
+
+  public keytipTree: KeytipTree = new KeytipTree();
+  public currentSequence: IKeySequence;
+  public keytips: IUniqueKeytip[] = [];
+
+  private _layer: KeytipLayerBase;
+
+  private _enableSequences: IKeytipTransitionKey[];
+  private _exitSequences: IKeytipTransitionKey[];
+  private _returnSequences: IKeytipTransitionKey[];
+  private _newCurrentKeytipSequences?: IKeySequence[];
+
+  private _keytipIDCounter = 0;
+
+  private _delayedKeytipQueue: string[] = [];
+  private _delayedQueueTimeout: number;
+  private _async: Async = new Async();
+
+  /**
+   * Static function to get singleton KeytipManager instance
+   *
+   * @returns {KeytipManager} Singleton KeytipManager instance
+   */
+  public static getInstance(): KeytipManager {
+    return this._instance;
+  }
+
+  /**
+   * Sets the _layer property and other related variables
+   * Should be called from the KeytipLayer constructor
+   *
+   * @param layer - KeytipLayerBase object
+   */
+  public init(layer: KeytipLayerBase): void {
+    this._layer = layer;
+    this.currentSequence = '';
+    // All guaranteed to be set because of defaultProps in KeytipLayer
+    this._enableSequences = this._layer.props.keytipStartSequences!;
+    this._exitSequences = this._layer.props.keytipExitSequences!;
+    this._returnSequences = this._layer.props.keytipReturnSequences!;
+  }
+
+  /**
+   * Registers a keytip
+   *
+   * @param keytipProps - Keytip to register
+   * @returns {string} Unique ID for this keytip
+   */
+  public registerKeytip(keytipProps: IKeytipProps): string {
```

Comment:
>registerKeytip [](start = 9, length = 14)

seems that this function not only registers a new keytip but displays it an executes it if appropriate, the second part does not seem too related to register, should we break this in two or update the name of the function? registerAndShowIfNeeded or something like that?